### PR TITLE
Updates to CSV generation

### PR DESCRIPTION
**Pull request by kshlm**
This PR adds support for including extra manifests into the bundle when generating the unified CSV.

In addition there have been other small fixes to included manifests.

The CSV has been regenerated with a the rook/ceph:v1.0.0-526.g3ece503 with rook/rook#3725, noobaa/noobaa-operator:1.1.0 and ceph/ceph:v14.2.2-20190828

Also, the StorageCluster CR example has been updated to include monPVCTemplate.

### CODE REVIEW

**Comment by kshlm** [comment]
@davidvossel would be nice to have your review here.

**Comment by umangachapagain** [comment]
Why does it have the same changes as https://github.com/openshift/ocs-operator/pull/49 ?? It is already merged.

**Comment by kshlm** [comment]
> Why does it have the same changes as #49 ?? It is already merged.

If you look closely, it's not really the same changes.

**Comment by kshlm** [comment]
To test this, use the manifest athttps://gist.github.com/kshlm/0233ef67d5473f423fcb3ea37b90d062

```console
$ oc create -f https://gist.github.com/kshlm/0233ef67d5473f423fcb3ea37b90d062/raw/bdbba048d7d8763ad0720a84b566219584a28596/deploy-with-olm.yaml
```

File: deploy/bundlemanifests/role_binding_metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name: rook-ceph-metrics
+  namespace: openshift-storage
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: rook-ceph-metrics
+subjects:
+- kind: ServiceAccount
+  name: prometheus-k8s
+  namespace: openshift-monitoring
```
Comment (petr-balogh) [inline deploy/bundlemanifests/role_binding_metrics.yaml]: can we have new line at the end? I see that github shows red icon at the end of the line 13 which tells `No new line at end of file`

File: deploy/bundlemanifests/role_metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ - apiGroups:
+   - ""
+   resources:
+    - services
+    - endpoints
+    - pods
+   verbs:
+    - get
+    - list
+    - watch
```
Comment (petr-balogh) [inline deploy/bundlemanifests/role_metrics.yaml]: same

File: deploy/crds/ocs_v1alpha1_storagecluster_cr.yaml [hunk 3]
```diff
@@ -18,5 +26,5 @@ spec:
         volumeMode: Block
         resources:
           requests:
-            storage: 1Ti
```
[the remark this answers is not in the document]
Comment (kshlm) [reply deploy/crds/ocs_v1alpha1_storagecluster_cr.yaml]: This is just an example. We don't expect anyone to actually use it in production. And this saves on resources during testing.

**Comment by davidvossel** [comment]
> @davidvossel would be nice to have your review here.

hey, sorry i didn't get a chance to review this before it was merged. I do see a few conflicts.

Below is what we need to do. 

1. lock down the exact versions of the source containers we're using to generate the CSV. Right now there are subtle changes to the generated CSV when I generate it on my local machine. 

2. Never edit the CSV directly. Those changes get lost when the csv is generated.

3. We need a CI job to detect when the CSV generation results in uncommitted changes.

I'm working on PRs to address all of these. They should be posted by the end of the day.

**Pull request merged**